pinecone/provider/index_resource.go

```diff
@@ -301,7 +508,7 @@ func (r *IndexResource) Create(ctx context.Context, req resource.CreateRequest,
 		if err != nil {
 			return retry.NonRetryableError(err)
 		}
-		if !index.Status.Ready {
+		if !index.Status.Ready && index.Status.State != "Ready" {
```

Comment:
Does this really need to be retried? If something was created but isn't ready, seems like waiting is all that's needed. Although I've never seen a non-ready response from creating a serverless index so maybe this isn't even a real use scenario.